subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1508,6 +1554,10 @@ static int consume_pgps_data(uint8_t pnum, const char *buf, size_t buf_len)
 					evt_handler(&evt);
 				}
 			} else {
+#if defined(CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL)
+				k_sem_give(&flash_accessible);
+				LOG_DBG("flash_accessible UNLOCKED");
+#endif
```

Comment:
I can't spot the corresponding lock for this unlock

Reply:
It's in `open_storage()`, when the download begins.  This unlock is in `consume_pgps_data()`, when download occurs we've reached the end of the file.